include/net/tcp.h

```diff
@@ -27,9 +29,18 @@ struct tcphdr
     uint16_t check;   // Checksum
     uint16_t urg_ptr; // Urgent Pointer
 };
+#pragma pack(pop)
 
-// Verify the bit fields give the correct header size.
-#ifndef C_ASSERT
-#define C_ASSERT(e) typedef char __C_ASSERT__[(e) ? 1 : -1]
```

Comment:
This is ok, but I wonder if there is a reason to define C_ASSERT as static_assert so other code need not change?

Reply:
C_ASSERT doesn't have a human readable string attached to it, while static assert does. Instead, C_ASSERT produces a hard-to-understand failure at compile time vs a more human readable error.